drivers/accel/adxl38x/adxl38x.c

```diff
@@ -0,0 +1,992 @@
+			 uint16_t *raw_x, uint16_t *raw_y,
+			 uint16_t *raw_z, uint16_t *raw_temp)
+{
+	uint8_t array_raw_data[8] = {0};
+	uint8_t array_rearranged_data[8] = {0};
+	int ret;
+	int j = 0;
+	uint8_t write_data;
+	uint8_t op_mode;
+	uint8_t dig_en;
+	uint8_t tzyx, start_addr;
+	uint16_t num_bytes = 0;
+
+	ret = adxl38x_read_device_data(dev, ADXL38x_DIG_EN, 1, &dig_en);
+	write_data = dig_en;
+	write_data = write_data & ~ADXL38x_MASK_CHEN_DIG_EN;
+	write_data = write_data | ((channels << 4) & ADXL38x_MASK_CHEN_DIG_EN);
+
+	ret = adxl38x_write_device_data(dev, ADXL38x_DIG_EN, 1, &write_data);
+	if (ret)
+		return ret;
+	//Put part in measurement mode if not placed in it already
+	ret = adxl38x_read_device_data(dev, ADXL38x_OP_MODE, 1, &op_mode);
+	ret = adxl38x_get_op_mode(dev, &op_mode);
+	if(op_mode == 0) {
+		adxl38x_set_op_mode(dev, ADXL38x_MODE_HP);
+	} else {
+		adxl38x_set_op_mode(dev, op_mode);
+	}
+
+	//Find the channels and number of bytes to read
+	tzyx = (uint8_t)channels;
+	if(tzyx & 0x01) {
+		start_addr = ADXL38x_XDATA_H;
+	} else if(tzyx & 0x02) {
+		start_addr = ADXL38x_YDATA_H;
+	} else if(tzyx & 0x04) {
+		start_addr = ADXL38x_ZDATA_H;
+	} else {
+		start_addr = ADXL38x_TDATA_H;
+	}
+	while(tzyx) {
+		num_bytes += tzyx & 0x01;
+		tzyx >>= 1;
+	}
+	num_bytes = num_bytes * 2;
+
+	//Read accel data channels
+	ret = adxl38x_read_device_data(dev, start_addr, num_bytes, array_raw_data);
+	if (ret)
+		return ret;
+	//Re-assign raw data to corresponding channels
+	tzyx = (uint8_t)channels;
+	for(int i = 0; i <  8; i+=2) {
+		if(tzyx & 0x01) {
+			array_rearranged_data[i] = array_raw_data[j];
+			array_rearranged_data[i+1] = array_raw_data[j+1];
+			j += 2;
+		} else {
+			array_rearranged_data[i] = NULL;
+			array_rearranged_data[i+1] = NULL;
+		}
+		tzyx >>= 1;
+	}
+	*raw_x = adxl38x_accel_array_conv(array_rearranged_data);
+	*raw_y = adxl38x_accel_array_conv(array_rearranged_data + 2);
+	*raw_z = adxl38x_accel_array_conv(array_rearranged_data + 4);
+	*raw_temp = adxl38x_accel_array_conv(array_rearranged_data + 6);
+
+	return ret;
+}
+
+
+/***************************************************************************//**
+ * @brief Reads the raw output data of each axis and converts it to g.
+ *
+ * @param dev  - The device structure.
+ * @param channels - Channels to enable.
+ * @param x    - X-axis's output data.
+ * @param y    - Y-axis's output data.
+ * @param z    - Z-axis's output data.
+ *
+ * @return ret - Result of the reading procedure.
+*******************************************************************************/
+int adxl38x_get_xyz_gees(struct adxl38x_dev *dev,
+			 enum adxl38x_ch_select channels,
+			 struct adxl38x_fractional_val *x, struct adxl38x_fractional_val *y,
+			 struct adxl38x_fractional_val *z)
+{
+	uint16_t raw_accel_x;
+	uint16_t raw_accel_y;
+	uint16_t raw_accel_z;
+	uint16_t raw_accel_t;
+	uint16_t dum;
+	int ret;
+
+	int64_t dummy;
+
+	ret = adxl38x_get_raw_data(dev, channels, &raw_accel_x, &raw_accel_y,
```

Comment:
I would rather use adxl38x_get_raw_xyz and simply remove enum adxl38x_ch_select channels parameter and provide the data for all 3 channels.

Reply:
Based on requirement from Applications team